source/extensions/filters/common/ext_authz/ext_authz.h

```diff
@@ -27,6 +27,24 @@ enum class CheckStatus {
   Denied
 };
 
+typedef std::vector<std::pair<Http::LowerCaseString, std::string>> KeyValueHeaders;
+
+/**
+ * Authorization response object for a RequestCallback.
+ */
+struct Response {
+  // Call status.
+  CheckStatus status;
+  // Optional http headers attribute.
+  KeyValueHeaders headers;
```

Comment:
I think this could be HeaderMapPtr.

Reply:
If it makes thing easier or more efficient, sure.

Reply:
I think both if possible to move it later. I'll need to take a closer look at the HeaderMap api though.